src/yunohost/app.py

```diff
@@ -1265,60 +1245,91 @@ def _sanitized_absolute_url(perm_url):
             return perm_domain + perm_path
 
         # Skipped
-        skipped_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'skipped_uris')]
-        skipped_regex += _get_setting(app_settings, 'skipped_regex')
-
-        # Redirected
-        redirected_urls.update(app_settings.get('redirected_urls', {}))
-        redirected_regex.update(app_settings.get('redirected_regex', {}))
+        skipped_urls = [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'skipped_uris')]
+        skipped_urls += ['re:' + regex for regex in _get_setting(app_settings, 'skipped_regex')]
 
         # Legacy permission system using (un)protected_uris and _regex managed in app settings...
-        unprotected_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'unprotected_uris')]
-        protected_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'protected_uris')]
-        unprotected_regex += _get_setting(app_settings, 'unprotected_regex')
-        protected_regex += _get_setting(app_settings, 'protected_regex')
+        unprotected_urls = [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'unprotected_uris')]
+        protected_urls = [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'protected_uris')]
+        unprotected_urls += ['re:' + regex for regex in _get_setting(app_settings, 'unprotected_regex')]
+        protected_urls += ['re:' + regex for regex in _get_setting(app_settings, 'protected_regex')]
+
+        if skipped_urls == [] and unprotected_urls == [] and protected_urls == []:
+            continue
 
-        # New permission system
+        # Manage compatibility with old protected, unprotected, skipped urls !!
         this_app_perms = {name: info for name, info in all_permissions.items() if name.startswith(app + ".")}
         for perm_name, perm_info in this_app_perms.items():
 
             # Ignore permissions for which there's no url defined
             if not perm_info["url"]:
                 continue
 
-            # FIXME : gotta handle regex-urls here... meh
             url = _sanitized_absolute_url(perm_info["url"])
             perm_info["url"] = url
             if "visitors" in perm_info["allowed"]:
-                if url not in unprotected_urls:
-                    unprotected_urls.append(url)
-
                 # Legacy stuff : we remove now protected-urls that might have been declared as unprotected earlier...
                 protected_urls = [u for u in protected_urls if u != url]
             else:
-                if url not in protected_urls:
-                    protected_urls.append(url)
-
                 # Legacy stuff : we remove now unprotected-urls / skipped-urls that might have been declared as protected earlier...
```

Comment:
Don't really understand why we do this. @alexAubin still it necessary ?

@alexAubin, do know if would be possible to migrate all `unprotected_urls`, `skipped_urls` and `protected_urls` to some permission. By example we could create something like that:
permissions:
- `app.legacy_skipped_uris`: list of all `skipped_uris` and `skipped_regex`
- `app.legacy_unprotected_urls`: list of all `unprotected_urls` and `unprotected_regex`
- `app.legacy_protected_urls`: list of all `protected_urls` and `protected_regex`

Reply:
If I remember correctly, we do this because : 
- some legacy app can be installed and for example set (un)protected_uri to /
- a main permission is automatically created with the appropriate public/private (visitor allowed or not) state
- later, the user might change is visitors are allowed or not
- but this is not propagated on the (un)protected setting
- so right here we have to remove "some stuff that was (un)protected when looking at the legacy ... but should actually be different once we load the new permission system"

Of course you will say "that's batshit crazy, let's just migrate the legacy settings once and for all?" and you are right. But of course the complex questions start happening when you think about the fact that : 
- the settings might be more complex than just `/`
- legacy apps may re-write the (un)protected setting during upgrades/restore ...
- ??? other shitty case that I may have forgot about ...

Reply:
![index](https://user-images.githubusercontent.com/4533074/78249306-6abc8680-74ee-11ea-8e26-9cc05227bece.jpeg)

Reply:
Thanks for you answer.

> * the settings might be more complex than just `/`

For this I don't know how to manage this. Do you have an example.

> * legacy apps may re-write the (un)protected setting during upgrades/restore ...

Maybe we can do something in the helper for this ?

Reply:
We could replace all (not exactly the syntax but it's for the idea):
`ynh_app_settings_set $app uprotected_uris "url"`
by
`ynh_permission_url "main" url` and `ynh_permission_update "main" "visitors"`?

Reply:
>For this I don't know how to manage this. Do you have an example.

https://github.com/YunoHost-Apps/etherpad_mypads_ynh/blob/testing/scripts/install#L314

or

https://github.com/YunoHost-Apps/ttrss_ynh/blob/testing/scripts/install#L163

(But granted that the vast majority of apps just play with '/')

>Maybe we can do something in the helper for this ?

Hmyea indeed, we could try to something like `if (current_script != install) and (there's already a permission that corresponds to the url) then [ ignore the change ]` 

(again, not sure about all the detail and edge cases...)

Reply:
>We could replace all (not exactly the syntax but it's for the idea):
ynh_app_settings_set $app uprotected_uris "url"
by
ynh_permission_url "main" url and ynh_permission_update "main" "visitors"?

Yes and no ... For example, if url = "/admin", you don't want to define this as the url of the main permission ... A new permission has to be created, and it feels weird to automatically do this for the packager.

And if url = "a list" you are screwed...

Reply:
Tried to migrate all of theses things here https://github.com/YunoHost/yunohost/pull/861/commits/9cdf00b0315daaebc332ad7cd28554a97d712cd1 ...

Hope that I won't break everything...

Reply:
(Flagging this comment as resolved, feeling a bit lazy to re-read the whole thing, I think it was addressed by Josue's PR about the legacy permission PR thingy that got merged recently)